### DISCUSSION TITLE
Add support for Inspect( MonoBehavior[] scripts )

### DISCUSSION THREAD

**Discussion by nukadelic** [yasirkula/UnityRuntimeInspector#32] [Ideas]
I'm using the inspector as an runtime alternative for an quick UI alternative for my math demos , right now I'm inspecting a single MonoBehavior, and would like to have the ability to pass array as an argument , It seems like gameObjects also work but it will display unnecessary information and controls that the demo shouldn't have. Also would be awesome to have the component `Enabled` checkbox be optional as well as the component collapse options as optional in the `RuntimeInspector` settings ( similar to how ShowTooltips works rn )  

Here is a preview of what I have right now, the top part is crossed out as its not needed 

![image](https://i.imgur.com/MDnuX6E.png)

**Comment by nukadelic** [comment]
I managed to hide `Enabled` check box by making a clone of the inspector settings SO , but still can't remove `Transform` and `GameObject` , nor I can see where to make the collapse arrow optional . 
![image](https://i.imgur.com/vlHxOxy.png)

**Comment by yasirkula** [comment]
Using a **RuntimeInspectorCustomEditor** would probably be the fastest solution here. First, declare the following classes:

```csharp
public class ComponentsWrapper
{
	public IList<Component> components;
}

[RuntimeInspectorCustomEditor( typeof( ComponentsWrapper ), false )]
public class ComponentsWrapperEditor : IRuntimeInspectorCustomEditor
{
	public void GenerateElements( ObjectField parent )
	{
		// Get the components we are inspecting
		IList<Component> components = ( (ComponentsWrapper) parent.Value ).components;
		for( int i = 0; i < components.Count; i++ )
		{
			Component component = components[i];
			if( component is Transform )
				continue;

			// Create drawer for component
			ExpandableInspectorField componentField = (ExpandableInspectorField) parent.CreateDrawerForComponent( component );

			// Don't show arrow for component
			componentField.HeaderVisibility = RuntimeInspector.HeaderVisibility.AlwaysVisible;
		}
	}

	public void Refresh() { }
	public void Cleanup() { }
}
```

Then, simply create an instance of ComponentsWrapper with the desired components and pass it to RuntimeInspector's Inspect function: `runtimeInspector.Inspect( new ComponentsWrapper() { components = desiredComponents } );`

**Reply by nukadelic** [reply]
![image](https://i.imgur.com/hUBYox6.png)
![image](https://i.imgur.com/YVuphuu.png)

**Comment by nukadelic** [comment]
Quick note for anyone who will be further modding the looks of the UI, there is an auto layout code that seems to override the padding on any custom skins we make ( I made a copy of the button prefab and only changed its `InvokeButton` relative size ), so instead of changing the source ( using git package via upm ) I just added a small script to set the rect transform values on component start so it would look something like this 

![image](https://i.imgur.com/fHpZkQb.png)

Thank you for making this FOSS

**Reply by yasirkula** [reply]
You are welcome. Alternatively, you can comment out these lines: https://github.com/yasirkula/UnityRuntimeInspector/blob/b1041184a24134c585522cf62542a967912cdc02/Plugins/RuntimeInspector/Scripts/RuntimeInspector/Fields/ExposedMethodField.cs#L33-L36
